src/zope/app/container/browser/tests/test_contents_functional.py

```diff
@@ -187,20 +184,18 @@ def test_inplace_change_title(self):
         response = self.publish('/@@contents.html',
                                 basic='mgr:mgrpw',
                                 form={'retitle_id': u'foo'})
-        body = ' '.join(response.getBody().split())
-        self.assert_(body.find('type="hidden" name="retitle_id"') >= 0)
-        self.assert_(body.find('input name="new_value"') >= 0)
-        self.assert_(body.find('type="submit" name="container_cancel_button"')
-                     >= 0)
-        self.assert_(body.find('type="submit" name="container_rename_button"')
-                     < 0)
+        body = ' '.join(response.text.split())
+        self.assertIn('type="hidden" name="retitle_id"', body)
+        self.assertIn('input name="new_value"', body)
+        self.assertIn('type="submit" name="container_cancel_button"', body)
+        self.assertNotIn('type="submit" name="container_rename_button"', body)
```

Comment:
`assert_` is deprecated and triggers a deprecation warning. It was simple to replace them all with a regex once I thought about it.

But the main reason I did this was to be able to see the body text when the assertion failed, as it often did early in the porting process.

Hmm, that might be a disadvantage of a higher-level DOM based approach.